packages/node_modules/cerebral/src/utils.js

```diff
@@ -277,33 +277,16 @@ export function createDummyController(state = {}, signals = {}) {
 }
 
 export function copyAndFreezeObject(rootObject) {
-  function makeCopy(value) {
-    if (Array.isArray(value)) {
-      return value.slice()
-    } else if (isObject(value)) {
-      return Object.assign({}, value)
-    }
-
-    return value
-  }
-
-  function freezeObject(object) {
-    if (!Object.isFrozen(object) && isComplexObject(object)) {
-      for (const key in object) {
-        // Properties might not be writable, but then there
-        // is not reason to freeze its value either
-        try {
-          object[key] = freezeObject(object[key])
-        } catch (e) {}
-      }
-
-      return Object.freeze(makeCopy(object))
+  function deepFreeze(val) {
+    if (isComplexObject(val)) {
+      Object.getOwnPropertyNames(val).forEach(name => {
+        deepFreeze(val[name])
+      })
+      return Object.freeze(val)
     }
-
-    return object
+    return val
   }
-
-  return freezeObject(rootObject)
+  return deepFreeze(JSON.parse(JSON.stringify(rootObject)))
```

Comment:
I think JSON will work fine as this will only happen using `state.get` inside actions. So it is pointing to a specific value. It is also only in development :)